### ISSUE TITLE
Non-English file names cause the debugger to choke

### ISSUE THREAD

**Issue by TradeIdeasPhilip** [denoland/vscode_deno#517]
**Describe the bug**

If I create a TypeScript file will non-English characters in the filename, the debugger gets confused.  As does Intellisense.

I'm pretty sure this is a new problem.  I've been using some of these file names for a while.

**To Reproduce**

_debugger_

1. Create a file with a name like `¡Feliz cumpleaños!.ts` or `⛔message-from-server.ts`.
2. Try to set a breakpoint in this file.
3. Initially (before you try to run) the breakpoint will appear as it should.
4. Run the program.
5. The breakpoint will disappear. 
6. Try to set the breakpoint again, now that the program is running.
7. It will disappear immediately
8. Set a breakpoint somewhere else, and step through your file.
9. When you get to this file, the debugger will show you the compiled version of the code, e.g. without the TypeScript annotations or the comments.
10. You _can_ set a breakpoint in the compiled file and that _will_ work.
11. As soon as you stop running the compiled file will disappear, so you can't easily access it again.

_intellisense_

1. Create a file with a name like `¡Feliz cumpleaños!.ts` or `⛔message-from-server.ts`.
2. In a different file, reference an item that is exported from the file with the non-English characters in the name.
3. Intellisense will suggest adding an import.  Accept its suggestion.
4. The line it creates will quote all of the non-English characters like this:  `import { celebrate1 } from "./%C2%A1Feliz%20cumplea%C3%B1os!.ts";`
5. You can keep that ugly file name with the URL-style quotes or you can change it to the easy to read format.  It doesn't make a difference.
6. Either way the program will run fine.
7. Either way the debugger will act up like described above.

**Expected behavior**

VS Code should allow you to use most characters in file names. (Like the TypeScript compiler and the JavaScript runtime.)

The debugger should work like normal.  I should be able to set breakpoints in any file.  I should be able to see my comments and my TypeScript annotations while I'm stepping though my code.

The suggested line should not try to quote the file name.  It's already in double quotes, so you don't need the URL quoting on top of that.  

Note that the file export it adds includes the `.ts` file extension.  So I know the Deno extension is involved.  Without the Deno extension VS code would remove the file extension before trying to add the import.

**Screenshots**

Create a file with non-English characters in the file name:
![Screenshot (320)](https://user-images.githubusercontent.com/18409827/132911017-e2d8f958-e12b-4ea2-83c4-fa21097a63ca.png)

Reference this file from another file.  Notice the intellisense recommendation:
![Screenshot (319)](https://user-images.githubusercontent.com/18409827/132911321-18678fc1-d52f-4d67-b306-3ccb1c0fd85d.png)

Try to set a breakpoint on line 4 of the file:
![image](https://user-images.githubusercontent.com/18409827/132911589-693f55b6-e8e5-4fe2-bc16-4032b191e4c7.png)

When you step through the code you'll see the the compiled version of the code.  Notice that line 4 of the compiled file has a breakpoint, but it does not correspond to line 4 of the source file:
![image](https://user-images.githubusercontent.com/18409827/132911848-a38ce9ed-17b8-4d3b-84ea-2af7471266c6.png)

**Versions**

vscode: 1.60.0 deno: xxx extension: v3.8.1

Version: 1.60.0 (user setup)
Commit: e7d7e9a9348e6a8cc8c03f877d39cb72e5dfb1ff
Date: 2021-09-01T10:41:52.311Z
Electron: 13.1.8
Chrome: 91.0.4472.164
Node.js: 14.16.0
V8: 9.1.269.39-electron.0
OS: Windows_NT x64 10.0.19042

I noticed this shortly after VS Code updated itself.  I know VS Code did an update because of an [unrelated bug](https://github.com/microsoft/vscode/issues/132150).  The Deno extension updates itself automatically, so I don't know if that's new or not.

**Comment by kitsonk** [comment]
Could you provide the output of _View | Output_ and then select _Deno Language Server_?

**Comment by TradeIdeasPhilip** [comment]
Starting Deno language server...
  version: 1.13.2 (release, x86_64-pc-windows-msvc)
  executable: C:\Users\Philip Smolen\.deno\bin\deno.EXE
Connected to "Visual Studio Code" 1.60.0
Enabling import suggestions for: https://deno.land
language server initialized
Server ready.

**Comment by kitsonk** [comment]
Thank you. It is a bug, I don't think we anticipated it.  There are two underlying issues, the debugger and the code actions (the import suggestion).  Like there are other issues.  We need to deal with some URLs that contain URL encoded characters.

**Comment by TradeIdeasPhilip** [comment]
I saw an update to the other bug report I referenced.  https://github.com/microsoft/vscode/issues/132150#issuecomment-918480453 

❝I left for vacation and came back to my entire JavaScript debugging setup being borked beyond use. Not only is it opening PowerShell for debug sessions with "integratedTerminal" option, but **it is now hitting breakpoints on a new tab to my (already opened) source file in read-only mode**. What a mess.❞

The part in **bold** sounds similar to one of the problems that I mentioned in this bug report.  It's hard for me to say if the VS Code bug is responsible for the problem I described, or if it's the Deno extension.
